packages/lib-runtime/src/store/useExtensions.ts

```diff
@@ -20,7 +24,7 @@ const getData = (pluginConsumer: PluginConsumer) => pluginConsumer.getExtensions
 export const useExtensions = <TExtension extends Extension>(
   predicate?: ExtensionPredicate<TExtension>,
 ): LoadedExtension<TExtension>[] => {
-  const extensions = usePluginSubscription(eventTypes, getData);
+  const extensions = usePluginSubscription(eventTypes, getData, isSameData);
```

Comment:
This change is needed since `pluginConsumer.getExtensions()` always returns a new array.

We want to do reference comparison on extensions themselves, not on the array containing them.